python/ml4ir/base/model/scoring/scoring_model.py

```diff
@@ -303,13 +303,18 @@ def __update_metrics(self, inputs, y_true, y_pred):
         - Currently only support pre-compiled Keras metrics
         """
         # Compute metrics on primary label
-        self.compiled_metrics.update_state(y_true, y_pred)
+        try:
+            mask = inputs[self.feature_config.get_mask("node_name")]
+        except KeyError:
```

Comment:
You can do `except KeyError, AttributeError:` @mohazahran

Reply:
python refused to do so that's why I spread it into two